src/main/kotlin/graphics/scenery/volumes/Volume.kt

```diff
@@ -1221,14 +1221,17 @@ open class Volume(
          * into account.
          */
         override fun composeModel() {
-            val shift = Vector3f(volume.getDimensions()) * (-0.5f)
-
             model.translation(position)
             model.mul(Matrix4f().set(this.rotation))
             model.scale(scale)
             model.scale(volume.localScale())
-            if (volume.origin == Origin.Center) {
-                model.translate(shift)
+            when(val o = volume.origin) {
+                is Origin.Center -> {
+                    val shift = Vector3f(volume.getDimensions()) * (-0.5f)
+                    model.translate(shift)
+                }
+                is Origin.Custom -> model.translate(o.origin)
+                else -> {}
```

Comment:
what about FrontBottomLeft? Why is this not taken into account here?

Reply:
The Origins class only has the most common options in there, which is center and FrontBottomLeft. Further adjustments can be made with the Custom choice 👍